project_overview.md

```diff
@@ -0,0 +1,64 @@
+  * Secondary application: Sometimes it is useful to have an auxiliary program running besides the main application. This option currently only supports `Keepass`.
+  * Picture token (mandatory) for identifying the Shark Cage instance when run (picture is also displayed inside the secure desktop by the `CageLabeler`). This can either be from the users storage or directly captured by a connectec webcam.
+  * Name of configuration (mandatory). This name is used for the name of the config file and displayed by the `CageChooser`.
+
+## CageManager
+
+* __base64__: De- and encodes plain text from / into base64.
+* __CageDesktop__: Responsible for creating and switching to a new desktop and reserving space for the `CageLabeler`. When the secure desktop is no longer needed, this class also switches back to the previous desktop.
+* __CageLabeler__: Displays information about the currently running configuration: picture token, application names and hashes, activation buttons for the applications (to bring them in focus or restart them), exit restriction information, exit button.
+* __CageManager__: Immediately after getting started the `CageManager` listens for a tcp message. If no message arrives it just exits without doing anything else. If there is an expected message it parses it and then validates (signature / hash) all binaries specified in the reveived configuration. After validation it creates the security attributes with the `SecuritySetup`, creates a new access token with the `tokenLib` and a new desktop with a unique name & the security attributes. After this, the `CageLabeler` is used to display information on this new desktop. Once all this preparation is done the actual processes get started on the secure desktop and the `CageManager` waits for the correct exit conditions before cleaning up and exiting itself.
+* __FullWorkArea__: Used by the `CageDesktop` class to initialize and reserve space for the `CageLabeler`.`
+* __SecuritySetup__: Sets up and populates a [SECURITY_ATTRIBUTES](https://msdn.microsoft.com/en-us/library/windows/desktop/aa379560%28v=vs.85%29.aspx?f=255&MSPPError=-2147217396) struct used for example to create the secure desktop.
+
+## CageService
+
+* __CageService__: Checks if configuration files gotten as messages from `CageChooser` (and passed on by `CageServiceMain`) have the right access rights, asserts if the `CageManager` binary has the correct certificate and then starts it in the context of the current user (passing on the configuration file as well). Afterwards it waits until the `CageManager` instance closes again and only then accepts further messages (ensuring only one secure desktop can run at the same time).
+* __CageServiceMain__: Starts the `CageService`, continuously waits for messages to arrive (& pass them to the `CageService` instance) and waits for shutdown event in case the windows service gets stopped.
```

Comment:
Same as above: ...arrive (~&~ and pass them...

Reply:
Again, did this to not write "and" too often. Still want me to change it?

Reply:
It's okay the way it is (in both cases).